test/jsonInt64.spec.js

```diff
@@ -1,22 +1,69 @@
-var mocha  = require('mocha');
-var assert = require('chai').assert;
 var expect = require('chai').expect;
+var Assertion = require('chai').Assertion;
 var Int64 = require('node-int64');
 var JSONInt64 = require('../index');
 
-describe("Tests the Int64 support", function(){
+Assertion.addMethod('equalInt64', function (other)
+{
+    var object = this._obj;
+    new Assertion(object).to.be.an.instanceof(Int64);
+    new Assertion(other).to.be.an.instanceof(Int64);
+    this.assert(
+        object.equals(other),
+        "expected #{this} to be equal to #{act}",
+        "expected #{this} to be not equal to #{act}",
+        other,
+        object);
+});
+
+describe("Tests the Int64 support", function ()
+{
     var input = '{"int64MaxValue":9223372036854775807,"int64MinValue":-9223372036854775807,"number":123}';
 
-    it("tests the JSON parse and stringify functionality", function(done){
+    it("tests the JSON parse and stringify functionality", function (done)
+    {
         var parsedObject = JSONInt64.parse(input);
-        expect(parsedObject.number.toString()).to.equal("123");
+        expect(parsedObject.number).to.be.equal(123);
         expect(parsedObject.int64MaxValue).to.be.instanceof(Int64);
-        expect(JSONInt64.util.toDecimalString(parsedObject.int64MaxValue)).to.equal("9223372036854775807");
+        expect(JSONInt64.toDecimalString(parsedObject.int64MaxValue)).to.be.equal("9223372036854775807");
         expect(parsedObject.int64MinValue).to.be.instanceof(Int64);
-        expect(JSONInt64.util.toDecimalString(parsedObject.int64MinValue)).to.equal("-9223372036854775807");
+        expect(JSONInt64.toDecimalString(parsedObject.int64MinValue)).to.be.equal("-9223372036854775807");
 
         var expectedObject = JSONInt64.stringify(parsedObject);
-        expect(expectedObject).to.equal(input);
+        expect(expectedObject).to.be.equal(input);
+        done();
+    });
+
+    it("tests the from/to decimal string conversion utilities", function (done)
+    {
+        var decimalString = "1234";
+        var decimalInt64 = new Int64(1234);
+        var negativeDecimalString = "-1234";
+        var negativeDecimalInt64 = new Int64(-1234);
+        var maxJsSafeIntegerPlusOneDecimalString = "9007199254740992";
```

Comment:
use the constants from ``Number`` whenever possible

Reply:
We cannot use them here since `Number.MAX_SAFE_INTEGER + 1` would be outside of what can be representable by a javascript `Number`

Also this tests the conversion from `Int64` to `String` and vice versa so there should be no `Number` involved.